### PR TITLE
Fix duplicate entries

### PR DESCRIPTION
**Pull request by mertakman**

### CODE REVIEW

File: heap_min.go [hunk 1]
```diff
@@ -2,32 +2,57 @@ package ch
...
 
-type minHeap []*minHeapVertex
+type MinHeap struct {
+	ids       []int64
+	distances map[int64]float64
+}
+
+func NewMinHeap() *MinHeap {
+	return &MinHeap{
+		ids:       make([]int64, 0),
```
Comment (magnushiie) [inline heap_min.go:17]: Is there a reason to use `make` here instead of `nil`?

**Review by magnushiie — APPROVED** [review]
This one looks good, but we should also fix the other heap in `heap_dist.go` - or better yet, remove the latter and make everything use `minHeap`

**Comment by magnushiie** [comment]
Also, let's move the second commit to another PR (and I doubt we should reuse it from the library and rather copy it, to have better locality).

File: heap_min.go (hunk 1 above)
Comment (mertakman) [reply heap_min.go:17]: Is there a reason to use nil here instead of make? :)

**Comment by mertakman** [comment]
> This one looks good, but we should also fix the other heap in `heap_dist.go` - or better yet, remove the latter and make everything use `minHeap`

lets make it on the future. for now , just fixing the current heap is good enough for this PR's scope

File: heap_min.go (hunk 1 above)
Comment (magnushiie) [reply heap_min.go:17]: `nil` is more [idiomatic](https://go.dev/wiki/CodeReviewComments#declaring-empty-slices).

**Pull request merged**